### ISSUE TITLE
Addon Policy support for IRSA-enabled Service Accounts

### ISSUE THREAD

**Issue by TBBle** [weaveworks/eksctl#2837]
**Why do you want this feature?**

I use [IAM Roles for Service Accounts](https://eksctl.io/usage/iamserviceaccounts/) rather than Instance Roles for better-constrained granting of permissions, and many of my IRSA `ServiceAccounts` are for features like Cluster Autoscaler, and eksctl already has an 'addon' describing the necessary policies for an _Instance Role_ to host that feature/application.

The use of IRSA is even more important for e.g., [Cluster Autoscaler on Fargate](https://github.com/kubernetes/autoscaler/issues/3008#issuecomment-717105714), as you are limited to one [Pod Execution Role](https://eksctl.io/usage/schema/#iam-fargatePodExecutionRoleARN) per fargate Profile, so you end up managing a bunch of Pod Execution Roles in eksctl, which _also_ don't support Addon Policies.

*Note*: I have not yet tested IRSA with Fargate myself, but running CA in Fargate is an improvement goal for my team's next EKS build, which is what reminded me of this issue. If IRSA doesn't work with Fargate, then I'd also want Addon Policy support for Fargate Profiles, although they don't even support `attachPolicy` so that is probably more work in eksctl and worth a separate FR.

**What feature/behavior/change do you want?**

A new key like [`withAddonPolicies`](https://eksctl.io/usage/schema/#nodeGroups-iam-withAddonPolicies) in the [`serviceAccounts` array](https://eksctl.io/usage/schema/#iam-serviceAccounts) is what I would expect to see.

I also suspect that `withAddonPolicy`, taking only a single policy, might be better, since using IRSA you probably only want one built-in policy per `ServiceAccount` anyway, as the whole point of IRSA is that you can run, e.g. Cluster Autoscaler and External DNS, each with their own `ServiceAccount` with only the IAM permissions _it_ needs.

```yaml
iam:
  serviceAccounts:
  - metadata:
      name: cluster-autoscaler
      namespace: kube-system
    withAddon: autoScaler
```

is much more succinct _and_ descriptive than the [documented example](https://eksctl.io/usage/iamserviceaccounts/):

```yaml
iam:
  serviceAccounts:
  - metadata:
      name: cluster-autoscaler
      namespace: kube-system
    attachPolicy:
      Version: "2012-10-17"
      Statement:
      - Effect: Allow
        Action:
        - "autoscaling:DescribeAutoScalingGroups"
        - "autoscaling:DescribeAutoScalingInstances"
        - "autoscaling:DescribeLaunchConfigurations"
        - "autoscaling:DescribeTags"
        - "autoscaling:SetDesiredCapacity"
        - "autoscaling:TerminateInstanceInAutoScalingGroup"
        Resource: '*'
```

and will also be covered when eksctl is updated to fix bugs or missing permissions in an addon.

As a specific example of such a benefit, the [`autoScaler` addon policy](https://github.com/weaveworks/eksctl/blob/b48a14044f599801e626d2781b635fa19b19088c/pkg/cfn/builder/iam_helper.go#L43-L55) includes `ec2:DescribeLaunchTemplateVersions` which is missing from the example in the documentation repeated above, and this difference has existed since 1cc92ee3fdeae5c9744ae667ad2e672202849b3e in mid 2019.

---

I'm surprised this isn't already a feature request. I had intended to try and implement it myself over the last few months, but time slipped away, and I guess I forgot to lodge an FR at the time when I was actively feeling the pain of copying large policy documents out of the addons code or service documentation into my eksctl config.

**Comment by michaelbeaumont** [comment]
@TBBle What do you think about even doing away with the need to specify `namespace` and `name`?
Something like:

```
iam:
  serviceAccounts:
  - addon: autoScaler
```
or

```
iam:
  addonServiceAccounts:
  - autoScaler
```

**Comment by TBBle** [comment]
Personally, I'd rather keep the namespace and name options. Not all of these services are installed in kube-system or another fixed namespace, e.g. external-dns, and in some cases, the service account name is dictated by the service's Helm chart, e.g., AWS-EBS-CSI has _two_ service accounts it needs, one for the CSI Controller and one for the Snapshot Controller, and both names are burned into the Helm chart, or were last time I looked.

And then there may be considerations of cluster naming policies for service accounts, etc.

I just used 'addon' because that's what those policies are called in the instance roles. I don't actually like that name, as we have too many other uses of addon. k8s has two uses of "addon", one in the docs which just means [third-party controllers](https://kubernetes.io/docs/concepts/cluster-administration/addons/) and one which appears in the source tree for things like nodelocal-dns-cache, and is hence similar to the EKS Addon usage of "optional features that can be installed as part of a cluster rollout".

I am generally happy with the existing `iam.servicveAccounts` section, it'd just be nice to compress a long, canned `attachPolicy` into a `withKnownPolicy` or `withCannedPolicy` etc. option.

It shouldn't preclude `attachPolicy` for the same policy, as potentially eksctl's bundled policy might be out-of-date, so it should be possible to patch around that in the local config.
